vault/plugin_catalog.go

```diff
@@ -59,14 +104,213 @@ func (c *Core) setupPluginCatalog(ctx context.Context) error {
 	return nil
 }
 
+type pluginClientConn struct {
+	*grpc.ClientConn
+	id string
+}
+
+var _ grpc.ClientConnInterface = &pluginClientConn{}
+
+func (d *pluginClientConn) Invoke(ctx context.Context, method string, args interface{}, reply interface{}, opts ...grpc.CallOption) error {
+	// Inject ID to the context
+	md := metadata.Pairs(pluginutil.MultiplexingCtxKey, d.id)
+	idCtx := metadata.NewOutgoingContext(ctx, md)
+
+	return d.ClientConn.Invoke(idCtx, method, args, reply, opts...)
+}
+
+func (d *pluginClientConn) NewStream(ctx context.Context, desc *grpc.StreamDesc, method string, opts ...grpc.CallOption) (grpc.ClientStream, error) {
+	// Inject ID to the context
+	md := metadata.Pairs(pluginutil.MultiplexingCtxKey, d.id)
+	idCtx := metadata.NewOutgoingContext(ctx, md)
+
+	return d.ClientConn.NewStream(idCtx, desc, method, opts...)
+}
+
+func (p *pluginClient) Conn() grpc.ClientConnInterface {
+	return p.clientConn
+}
+
+// MultiplexingSupport determines if a plugin client supports multiplexing
+func (p *pluginClient) MultiplexingSupport(ctx context.Context) (bool, error) {
```

Comment:
Nit: Can we move this closer to where the struct is define?

Reply:
I am not sure we even need this anymore. After I pulled in Brian's change we are no longer using this method. Instead we are calling `pluginutil.MultiplexingSupported` directly in `isDatabasePlugin`.

Reply:
Which is why we were able to remove MultiplexingSupport() from the PluginClient interface.

Reply:
I went ahead and removed this method in favor of using `pluginutil.MultiplexingSupported` directly